src/main/java/org/folio/rest/helper/FundsHelper.java

```diff
@@ -100,23 +99,37 @@ public FundsHelper(Map<String, String> okapiHeaders, Context ctx, String lang) {
       });
   }
 
-  private CompletableFuture<String> getCurrentFiscalYearId(String ledgerId) {
+  public CompletableFuture<FiscalYear> getCurrentFiscalYear(String ledgerId) {
     FiscalYearsHelper fiscalYearsHelper = new FiscalYearsHelper(httpClient, okapiHeaders, ctx, lang);
+    return getTwoFirstFiscalYears(ledgerId, fiscalYearsHelper)
+      .thenApply(twoFirstFiscalYears -> {
+        if(CollectionUtils.isNotEmpty(twoFirstFiscalYears)) {
+          if(twoFirstFiscalYears.size() > 1 && isOverlapped(twoFirstFiscalYears.get(0), twoFirstFiscalYears.get(1))) {
+            return twoFirstFiscalYears.get(1);
+          } else {
+            return twoFirstFiscalYears.get(0);
+          }
+        } else {
+          return null;
+        }
+      });
+  }
+
+  private boolean isOverlapped(FiscalYear first, FiscalYear second) {
```

Comment:
Say we have overlapped FYs:
-  FY20: Dec 28 2019 - Dec 30th 2020
-  FY21: Dec 25 2020 - Dec 30th 2021
-  current date = Dec 27th 2019

According to 1st point in the story FY20 is expected, but `isOverlapped` will return true and the second FY will be returned.
I thin we have to check that the current date is in that overlapping period.